wfe2/wfe.go

```diff
@@ -2204,23 +2212,20 @@ func (wfe *WebFrontEndImpl) NewOrder(
 
 	// TODO(#5545): Spending and Refunding can be async until these rate limits
 	// are authoritative. This saves us from adding latency to each request.
-	doneCheckingLimits := make(chan struct{})
+	limitCheck := make(chan struct{})
 	var ratelimitTxns []ratelimits.Transaction
 	var newOrderSuccessful bool
 	var errIsRateLimit bool
 
 	go func() {
 		// Close the channel on goroutine completion.
-		defer close(doneCheckingLimits)
+		defer close(limitCheck)
 		ratelimitTxns = wfe.checkNewOrderLimits(ctx, acct.ID, names)
 	}()
 
 	defer func() {
-		// Wait for the rate limit check to complete before attempting to refund
-		// the limits. If the check failed, we don't want to refund anything.
-		<-doneCheckingLimits
-		if !newOrderSuccessful && !errIsRateLimit && ratelimitTxns != nil {
-			wfe.refundNewOrderLimits(ctx, ratelimitTxns)
+		if !newOrderSuccessful && !errIsRateLimit {
+			go wfe.refundNewOrderLimits(ctx, limitCheck, ratelimitTxns)
```

Comment:
Two thoughts:
1) It's very likely that the refund will never succeed. The function isn't called until the parent function is exiting, and from there it's likely microseconds until the context is canceled due to the API returning. Probably a use-case for context.WithoutCancel and setting a separate deadline on the refund.
2) If the async checkNewOrderLimits takes a long time for some reason -- longer than the rest of the WFE's newOrder function -- then both of these bools will still have their default value `false`, and `ratelimitTxns` will be nil. That may not result in expected behavior.

Reply:
Good points. I'm going to pull back on the added complexity here and embrace the simpler approach we had before.